### ISSUE TITLE
ScrollView contentContainerStyle

### ISSUE THREAD

**Issue by Iamsamwhoami** [Microsoft/reactxp#87]
Heya 👋 
I'm not getting `contentContainerStyle` to work on `<ScrollView />` is this even implemented?, it's mentioned in the docs. Also on web the `<ScrollView />` defaults to `display: block;` and setting display (to override it to "flex") in RN makes it error 😢

**Comment by erictraut** [comment]
contentContainerStyle is a bug in the documentation. We supported it for a while but removed it because it's unnecessary. You can simply add another <View> within the scroll view if you want to give it a unique style. I'll update the documentation.

The display property of the ScrollView is an internal implementation detail for ReactXP's ScrollView on the web. You shouldn't need to override it. As you've discovered, the "display" style isn't supported on RN because all layout is implicitly done using flexbox on RN. Out of curiosity, why did you want to override it?

**Comment by Iamsamwhoami** [comment]
@erictraut Hmm 🤔  maybe I'm thinking wrong but if the element isn't `display: flex;` we can't use `alignItems: center;`, `justifyContent: center;` to center the content inside the ScrollView?

Basically I want to do a login view where the keyboard pops up and you are able to scroll if the content get's bigger than screen real estate.

**Comment by erictraut** [comment]
Ah, I see. Just place a container (a "View") within the ScrollView, then place your other elements within that container. I think that will do what you want.

**Comment by Iamsamwhoami** [comment]
@erictraut Not sure how I can get my child view to take "100%" of the parent when the parent is a `display: block;` tho?

**Comment by erictraut** [comment]
The default styling for RX.View should do the trick. I just confirmed this in the hello world app, and it works fine on both web and RN.

<RX.ScrollView>
   <RX.View>
       // Other views go here
   </RX.View>
</RX.ScrollView>

**Comment by Iamsamwhoami** [comment]
@erictraut I don't manage to get that to work (even with zero styles) this is what I want to achieve:

<img width="1269" alt="screen shot 2017-05-12 at 08 34 25" src="https://cloud.githubusercontent.com/assets/5459507/25986007/f591f41c-36ed-11e7-8f32-ef1f6ff8c526.png">

(looks like this when I set `display: flex;` on the ScrollView)

This is what I get:

<img width="1272" alt="screen shot 2017-05-12 at 08 34 40" src="https://cloud.githubusercontent.com/assets/5459507/25986026/fd4f8a66-36ed-11e7-8331-6946a3f89264.png">

**Comment by erictraut** [comment]
@Iamsamwhoami Yes, you're right, this is a problem. The RN and web versions are consistent, but neither of them do what you want them to. I think I have a simple fix. I'll test it out and get it published in the next few days.

**Comment by erictraut** [comment]
@Iamsamwhoami I spent a couple more hours on this problem, but I wasn't able to get it to work consistently on both React Native and on the web. There's probably a way to do it, but I can't afford to invest more time on this.

What you're trying to do is quite unusual. I've used (and even written) many UI frameworks over the years, and all cases I can remember, the content height of a scroll view is always independent of the viewport height. In other words, the content that's being displayed within the scrolling region is not influenced by the height of the viewport. In your case, you want the effective content height to be equal to min(login box height, viewport height), and you want the content (the login box) to then be centered within the content region.

Here are some alternative solutions for you:
1. Remove the ScrollView. I'm not sure why you think a ScrollView is necessary in this case anyway. Your login box isn't that high, so it should fit vertically within any reasonable window/screen size. If you remove the ScrollView, it's easy to center the box.
2. Place the login box within a View and center it. Then use an onLayout callback to measure the height of your login box and the ScrollView height and manually compute the min(login-box-height, viewport height) and set the height of the containing View to that height.
